addons/binding/org.openhab.binding.bigassfan/src/main/java/org/openhab/binding/bigassfan/discovery/BigAssFanDiscoveryService.java

```diff
@@ -0,0 +1,270 @@
+
+        Pattern pattern = Pattern.compile("[(](.*);DEVICE;ID;(.*);(.*)[)]");
+        Matcher matcher = pattern.matcher(device.getDiscoveryMessage());
+        if (matcher.find()) {
+            logger.debug("Match: grp1={}, grp2={}, grp(3)={}", matcher.group(1), matcher.group(2), matcher.group(3));
+
+            // Extract needed information from the discovery message
+            device.setLabel(matcher.group(1));
+            device.setMacAddress(matcher.group(2));
+            String[] modelParts = matcher.group(3).split(",");
+            if (modelParts.length != 2) {
+                logger.info("Unable to extract device type from discovery message");
+                return;
+            }
+            device.setType(modelParts[0]);
+            device.setModel(modelParts[1]);
+
+            deviceDiscovered(device);
+        }
+    }
+
+    private synchronized void deviceDiscovered(BigAssFanDevice device) {
+        logger.info("Device discovered: {}", device);
+
+        if (device.isSwitch()) {
+            logger.debug("Switches (controllers) are not handled currently");
+            return;
+        } else if (device.isFan()) {
+            logger.debug("Add fan with IP={}, MAC={}", device.getIpAddress(), device.getMacAddress());
+        } else {
+            logger.debug("Discovered unknown device type {} at IP={}", device.getModel(), device.getIpAddress());
+            return;
+        }
+
+        // We got a valid discovery message. Process it as a potential new thing
+        String serialNumber = device.getMacAddress().replace(":", "");
+
+        Map<String, Object> properties = new HashMap<>();
+        properties.put(THING_PROPERTY_MAC, device.getMacAddress());
+        properties.put(THING_PROPERTY_IP, device.getIpAddress());
+        properties.put(THING_PROPERTY_LABEL, device.getLabel());
+        properties.put(Thing.PROPERTY_SERIAL_NUMBER, serialNumber);
+        properties.put(Thing.PROPERTY_MODEL_ID, device.getModel());
+        properties.put(Thing.PROPERTY_VENDOR, "Haiku");
+
+        try {
+            ThingUID uid = new ThingUID(THING_TYPE_FAN, serialNumber);
+            if (uid != null) {
+                // If there's not a thing and it's not in the inbox, create the discovery result
+                if (callback != null && callback.getExistingDiscoveryResult(uid) == null
+                        && callback.getExistingThing(uid) == null) {
+
+                    logger.info("Creating discovery result for UID={}, IP={}", uid, device.getIpAddress());
+                    DiscoveryResult result = DiscoveryResultBuilder.create(uid).withProperties(properties)
+                            .withLabel(device.getLabel()).build();
+
+                    thingDiscovered(result);
+                } else {
+                    logger.debug("Thing or inbox entry already exists for UID={}, IP={}", uid, device.getIpAddress());
+                }
+            }
+        } catch (Exception e) {
```

Comment:
Yes, that might be true, but please do so only on the highest level, this method is indirectly called from line 61 and I spotted at least 3 levels where all Exceptions are caught, one catch RuntimeException should do the job. 

A RuntimeException Because every exception which you are not forced to catch by the compiler is a runtime exception.